Shapes/ShapeFactory.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Shapes
+{
+    /// <summary>
+    /// class reresents design pattern named factory to factorially create shapes 
+    /// </summary>
+    internal class ShapeFactory
+    {
+        /// <summary>
+        /// hash set represents the names of the shapes that are available
+        /// </summary>
+        public HashSet<string> shapesNames = new HashSet<string>()
+        {
+            "Rectangle",
+            "Square",
+            "Triangle",
+            "Rhombus",
+            "Parallelgram",
+            "Trapezium"
+        };
+
+        /// <summary>
+        /// util function that check if two string are equal ignoring the case
+        /// </summary>
+        /// <param name="s1">a string</param>
+        /// <param name="s2">another string</param>
+        /// <returns></returns>
+        private bool IsEqualIgnoreCases(string s1, string s2)
+        {
+            return string.Equals(s1, s2, StringComparison.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// factorially crreate a shape
+        /// </summary>
+        /// <param name="shapeType">a string of the name of the shape</param>
+        /// <returns>the returned shape</returns>
+        public Shape? GetShape(string shapeType)
+        {
+            if (shapeType == null)
```

Comment:
Why if else when all your if statements are the same boolean expression with different parameters? 
switch case!